src/Core/DocumentEvents.cs

```diff
@@ -54,8 +45,9 @@ public class Document(string fullPath, IEnumerable<AnalysisLanguage> detectedLan
 public interface IDocumentTracker
 {
     event EventHandler<DocumentEventArgs> DocumentClosed;
```

Comment:
While we're at this, why don't we unify this into a single event and differentiate by event arg types? This would make events linearized, which removes ambiguity.